ic-os/defs.bzl

```diff
@@ -553,17 +553,19 @@ EOF
         ],
         outs = ["launch_local_vm_script"],
         cmd = """
-        IMAGE="$(location :disk-img.tar)"
+        IMAGE="$$PWD/$(location :disk-img.tar)"
         cat <<EOF > $@
 #!/usr/bin/env bash
 set -euo pipefail
 cd "\\$$BUILD_WORKSPACE_DIRECTORY"
-TEMP=\\$$(mktemp -d)
+TEMP=\\$$(mktemp -d --suffix=.qemu-launch-remote-vm)
+# Clean up after ourselves when exiting.
+trap 'rm -rf \\$$TEMP' EXIT
 CID=\\$$((\\$$RANDOM + 3))
-cp $$IMAGE \\$$TEMP
 cd \\$$TEMP
-tar xf disk-img.tar
-qemu-system-x86_64 -machine type=q35,accel=kvm -enable-kvm -nographic -m 4G -bios /usr/share/ovmf/OVMF.fd -device vhost-vsock-pci,guest-cid=\\$$CID -drive file=disk.img,format=raw,if=virtio -netdev user,id=user.0,hostfwd=tcp::2222-:22 -device virtio-net,netdev=user.0
+tar xSf $$IMAGE
+truncate -s 128G target.img
+qemu-system-x86_64 -machine type=q35,accel=kvm -enable-kvm -nographic -m 4G -bios /usr/share/ovmf/OVMF.fd -device vhost-vsock-pci,guest-cid=\\$$CID -boot c -drive file=target.img,format=raw,if=virtio -drive file=disk.img,format=raw,if=virtio -netdev user,id=user.0,hostfwd=tcp::2222-:22 -device virtio-net,netdev=user.0
```

Comment:
https://github.com/dfinity/ic/pull/1451#discussion_r1786898444

> This same tooling is used for testing HostOS and GuestOS as well. If you'd like to add this for testing SetupOS, could you add a branch to the logic? We have a similar thing above for the remote VMs, in setting the `nested` flag.

Reply:
Node uses this to test GuestOS and HostOS, and this change breaks:

```
% bazel run ic-os/guestos/envs/dev:launch-local-vm-no-kvm

...

Gave up waiting for root file system device.  Common problems:
 - Boot args (cat /proc/cmdline)
   - Check rootdelay= (did the system wait long enough?)
 - Missing modules (cat /proc/modules; ls /dev)
ALERT!  /dev/vda5 does not exist.  Dropping to a shell!


BusyBox v1.36.1 (Ubuntu 1:1.36.1-6ubuntu3.1) built-in shell (ash)
Enter 'help' for a list of built-in commands.

(initramfs)
```

Reply:
I can see why.  I will fix very shortly.

Reply:
I believe I have successfully fixed the launch-local-vm targets for GuestOS and HostOS.  See commit 2e80d280d29931f4e57e08de20907ee9487d2daf .